velox/exec/fuzzer/AsyncDataCacheFuzzer.cpp

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/fuzzer/AsyncDataCacheFuzzer.h"
+
+#include <boost/random/uniform_int_distribution.hpp>
+
+#include <folly/executors/IOThreadPoolExecutor.h>
+#include "velox/common/caching/FileIds.h"
+#include "velox/common/caching/SsdCache.h"
+#include "velox/common/file/File.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/common/memory/Memory.h"
+#include "velox/common/memory/MmapAllocator.h"
+#include "velox/dwio/common/CachedBufferedInput.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+DEFINE_int32(steps, 10, "Number of plans to generate and test.");
+
+DEFINE_int32(
+    duration_sec,
+    0,
+    "For how long it should run (in seconds). If zero, "
+    "it executes exactly --steps iterations and exits.");
+
+DEFINE_int32(
+    max_num_reads,
+    100,
+    "Max number of reads to be performed per thread.");
+
+DEFINE_int32(num_threads, 16, "Number of threads to read.");
+
+DEFINE_int32(num_files, 8, "Number of data files to be created.");
+
+DEFINE_uint64(
+    offset_interval_bytes,
+    8 << 20,
+    "The offset bytes to be aligned at for cache reads.");
+
+DEFINE_uint64(
+    min_file_bytes,
+    32 << 20,
+    "Minimum file size in bytes of the data files to be created.");
+
+DEFINE_uint64(
+    max_file_bytes,
+    64 << 20,
+    "Maximum file size in bytes of the data files to be created.");
+
+DEFINE_int32(num_files_in_group, 3, "Number of files to be grouped together.");
+
+DEFINE_int64(memory_cache_bytes, 16 << 20, "Memory cache size in bytes.");
+
+DEFINE_uint64(ssd_cache_bytes, 128 << 20, "Ssd cache size in bytes.");
+
+DEFINE_int32(num_shards, 4, "Number of shards of SSD cache.");
+
+DEFINE_uint64(
+    ssd_checkpoint_interval_bytes,
+    64 << 20,
```

Comment:
I don't think we should expose those as parameter of fuzzer. They should be able to be randomized, for now if we want to keep it simple, we can make them constant

Reply:
Thanks for pointing that out. I've discussed with @xiaoxmeng and we'll decide later which parameters should be 1) randomized by fuzzer; 2) kept as configurable parameters; 3) fixed as constants. 

So I don't have a strong preference on how we should define them for now in this initial PR. Let's see if @xiaoxmeng has some different opinion.

Reply:
We can randomize this later if it is zero.